tests/applications/streaming_module_test.py

```diff
@@ -0,0 +1,115 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+from beamlime.applications._nexus_helpers import (
+    StreamModuleKey,
+    StreamModuleValue,
+    collect_streaming_modules,
+)
+
+
+def test_collect_streaming_modules_event_data(ymir: dict) -> None:
+    modules = collect_streaming_modules(ymir)
+    expected_ev44_modules = {
+        StreamModuleKey(
+            "ev44", "hypothetical_detector", f"ymir_{i}"
+        ): StreamModuleValue(
+            (
+                'entry',
+                'instrument',
+                f'hypothetical_detector_{i}',
+                'ymir_detector_events',
+            ),
+            {
+                "name": "ymir_detector_events",
+                "type": "group",
+                "children": [
+                    {
+                        "module": "ev44",
+                        "config": {
+                            "topic": "hypothetical_detector",
+                            "source": f"ymir_{i}",
+                        },
+                    }
+                ],
+                "attributes": [{"name": "NX_class", "values": "NXevent_data"}],
+            },
+        )
+        for i in range(2)
+    }
+    ev44_modules = {
+        key: value for key, value in modules.items() if key.module_type == "ev44"
+    }
+    assert ev44_modules == expected_ev44_modules
```

Comment:
Should we also test if there are errors here? `collect_streaming_modules` doesn't do any error handling.

Reply:
Sorry for late reply.

I checked the error testing/handling possibility but it seems like there should be proper validation step that confirms all the assumption that ``collect_streaming_modules`.
We should also have an integration test between `json` structure and `beamlime`.
It's a little out of the scope in this PR so I did it in #230 ...!